---
title: "Phylogenetic Comparative Methods"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Put the title of your vignette here}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---
  
# Phylogenetic Comparative Methods

## Introduction to phylogenies
Phylogenetic trees are branching diagrams illustrating the evolutionary relationships among taxa. 

* Each **node** represents a speciation event in evolution.

* A **branch** connects two nodes of the tree

* **Terminal Nodes** represent the data under comparison. Also known as **Operational Taxonomic Units** *(OTUs)*

# Introduction to Phylogenetic comparative methods (PCMs)
PCMs enable us to study the history of organismal evolution and diversification. With it we can explore in depth these topics:

* Evolutionary ties between organisms

* Find and understand relationships between an ancestral sequence and its descendants

* Estimate time of divergence between a group of organisms that share a common ancestor

## <code>APE</code> package and the <code>phylo</code> class
The <code>ape</code> package (Analysis of Phylogenetics and Evolution) is used within the community of evolutionary biologists for data analysis and as a framework for the development of new analytical methods.
```{r}
library(ape)
```

To read and load trees into R using ape, you can use the <code>read.nexus</code> or <code>read.tree</code> functions. The <code>read.nexus</code> function reads NEXUS formatted files while the <code>read.tree</code> function reads Newick formatted trees.
```{r}
tree <- read.nexus("./Potential Nexus Files/T6506.nex")
tree
```

When the tree is read in and processed, it creates a list of class phylo. This phylo type is the basis of most PCMs. 
```{r}
class(tree)
str(tree)
```

The <code>edge</code> variable in the tree contains the beginning and ending node number for the nodes and tips in the tree
```{r}
head(tree$edge)
```

The <code>tip.label</code> variable in the tree contains  the labels for the tips in the tree
```{r}
head(tree$tip.label)
```

<code>Nnode</code> contains the number of internal nodes in the tree
```{r}
head(tree$Nnode)
```

Graphical output in form of a cladogram can be obtained by calling <code>plot</code> with the tree as a parameter. This calls the <code>plot.phylo</code> function of the <code>ape</code> package. 
```{r}
plot(tree, cex = 0.5)
```
